test/query_dialog_test.rb

```diff
@@ -0,0 +1,113 @@
+#! /usr/bin/rspec
+# Copyright (c) 2014 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact Novell about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require_relative "spec_helper"
+require "systemd_journal/query_dialog"
+
+if !ENV["TRAVIS"]
+  describe SystemdJournal::QueryDialog do
+    def property(id, property)
+      Yast::UI.QueryWidget(Id(id), property)
+    end
+
+    # Sends a user input to libyui, optionally running some code before.
+    #
+    # @param value [Symbol/String] the input expected by UI.UserInput
+    # @param pre_hook [Proc] code used to tie the expectations or any other
+    #   action to input events. For example, checking the final status of the
+    #   dialog before closing it. The effect of an input can be checked on the
+    #   pre_hook of the next one to make sure libyui has already applied all the
+    #   changes.
+    def send_user_input(value, pre_hook: nil)
+      expect(Yast::UI).to receive(:UserInput) do
+        pre_hook.call if pre_hook
+        value
+      end
+    end
+
+    before(:all) do
+      Yast.ui_component = "ncurses"
+    end
+
+    before(:each) do
+      Yast::UI.OpenUI
+
+      # The QueryPresenter needs to check the number of previous boots
+      allow_to_execute(/journalctl --list-boots/)
+        .and_return(cmd_result_for("list-boots-11"))
+
+      # Ensure initial status of the dialog:
+      # all filter checkboxes unchecked...
+      Yast::UI.ChangeWidget(Id("unit"), :Value, false)
```

Comment:
@ancorgs I totally forgot, this is not supposed to work, since this code runs before `dialog.run` so before `OpenDialog` creates the widgets. We should recheck.